### PR TITLE
(#778) perf: add checking and loading scripts uniqueness in pipeline

### PR DESCRIPTION
**Pull request by akaNightmare**
https://github.com/luin/ioredis/issues/778

### CODE REVIEW

File: lib/pipeline.js [hunk 1]
```diff
@@ -268,6 +268,8 @@ Pipeline.prototype.exec = function (callback) {
 
   return this.redis.script('exists', scripts.map(function (item) {
     return item.sha;
+  }).filter(function (sha, index, shaList) {
+    return shaList.indexOf(sha) === index;
   })).then(function (results) {
```
Comment (luin) [inline lib/pipeline.js]:
```javascript
scripts.map(function (item) {
    return item.sha;
  }).filter(function (sha, index, shaList) {
    return shaList.indexOf(sha) === index;
  })
```

Would be better to change to

```javascript
Array.from(new Set(scripts.map(({sha}) => sha)))
```

File: test/functional/pipeline.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ describe('pipeline', function () {
         [null, 3],
         [null, '3']
       ]);
-      redis.disconnect()
+      redis.disconnect();
```
Comment (luin) [inline test/functional/pipeline.js]: Don't add `;` here. Code style changes should belong to a separate pull request.

File: lib/pipeline.js (hunk 1 above)
Comment (akaNightmare) [reply lib/pipeline.js]: resolved

File: test/functional/pipeline.js (hunk 2 above)
Comment (akaNightmare) [reply test/functional/pipeline.js]: resolved
